### PR TITLE
UI: Fix wrong language names

### PR DESCRIPTION
**Pull request by Vainock**
### Description
This renames a lot of languages to fix inconsistencies and wrong language names. My source is a [language mapping list by Mozilla](https://github.com/mozilla/language-mapping-list/blob/master/language-mapping-list.js).
I've also added the language `English (GB)` (which is going to be pushed with the next translation update) and renamed the English language to `English (US)`.

This PR gets also rid of the language string which was left over by https://github.com/obsproject/obs-studio/pull/2588 as we no longer need translators to verify the correctness of a language name.

### Motivation and Context
Wrong language names are bad.

### How Has This Been Tested?
Updated my locale.ini with this version and viewed the list of languages. I've also checked if everything works properly if you select the new `English (GB)` language as no `en-GB.ini` language files exist at this point.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by Vainock** [comment]
I've now update the two Chinese language names. Nevertheless were both previous versions not listed in the language map.
Chinese Simplified (China) is now just Chinese Simplified and Chinese Traditional (Taiwan) is now just Chinese Traditional.

File: UI/data/locale.ini [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ﻿[en-US]
-Name=English
+Name=English (USA)
```
Comment (RytoEX) [inline UI/data/locale.ini:2]: This is almost always "English (US)" in my experience.

File: UI/data/locale.ini [hunk 2]
```diff
@@ -155,13 +158,13 @@ Name=فارسی
...
 [ku-TR]
-Name=کوردی
+Name=Kurdî
 
 [pa-IN]
 Name=ਪੰਜਾਬੀ
 
 [az-Az]
-Name=Azərbaycanca
+Name=Azərbaycan dili
```
Comment (RytoEX) [inline UI/data/locale.ini:170]: End file with a newline.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Please see review comments.

**Comment by kkartaltepe** [comment]
These names seem inconsistent with what ive seen and what is available on public mozilla run sites.

https://developer.mozilla.org/en-US/docs/MDN/Getting_started
th-TH, ch-Hans, ch-Hant, pt-BR, hi-IN

are all rendered differently than the source you are using for reference.

**Comment by Vainock** [comment]
> These names seem inconsistent with what ive seen and what is available on public mozilla run sites.
> 
> https://developer.mozilla.org/en-US/docs/MDN/Getting_started
> th-TH, ch-Hans, ch-Hant, pt-BR, hi-IN
> 
> are all rendered differently than the source you are using for reference.

I'm not pretty sure what you're trying to say. Can you explain it in more detail?

**Comment by kkartaltepe** [comment]
> I'm not pretty sure what you're trying to say. Can you explain it in more detail?

This PR changes th-TH from "ไทย" to "ภาษาไทย", but if we look at mozillas own site under "language" you can see its rendered "ไทย"

Similarly for the other languages I mention.

**Comment by Vainock** [comment]
I've no why they don't comply with their own conventions, but I've checked again specifically for Thai as it seems like some languages have "language" in their name. So for example `Thai language` in Thai is just `Thai` in English.
See also [Wikipedia](https://en.wikipedia.org/wiki/List_of_language_names), it also says Thai is `ภาษาไทย`.
The other things you've mentioned aren't mistakes but rather other variants you can write this language in. E.g. `Português (Brasil)`, `Português do Brasil` or `Português (do Brasil)`.

**Comment by kkartaltepe** [comment]
By the same wikipedia article then sr-CR must be wrong too. lt-LT is inconsistent with https://en.wikipedia.org/wiki/Lithuanian_language etc etc.

Basically I dont really think there is much of a reason to change these unless something is brought up from a native speaker or there is some kind of intl standard of native representations of their languages and we are choosing specifically to adopt it wholesale.

Everything everywhere is inconsistent, so changing our names to match someone else's inconsistencies doesn't seem very beneficial. If you were a native speaker making a change for your language it might make more sense but I'm assuming you are not native in 20 odd languages.

**Comment by Vainock** [comment]
But if everything is inconsistent and nothing is really official, you have to stick with a variant. The best thing we could do now is I think just adopt the variants from Wikipedia as these language names are put there very likely by native speakers.
Also I think Wikipedia is more reliable than a 3 year old list made by only a hand full of Mozilla contributors.

What do you think about that?

**Comment by kkartaltepe** [comment]
If everything is inconsistent the best thing to do is keep everything consistent for our users and not change anything unless there is a very good reason (Obviously unrecognizable to a native speakers, etc). That was my point.

Personally I dont see a ton of value in any of these changes. The variants already used in OBS are pretty popular across various websites and localizers have already been over all of them (afaik). If localizers have no issues with them I see no reason to change them.

**Comment by RytoEX** [comment]
I'm reaching out to some native speakers with some questions. If the original source justifying this has now been called into question, I'd like to see new justification and a source for why these changes should be made.

**Comment by Vainock** [comment]
Okay, that would be great.
I mean, in the end these are the only people that can verify what is true and what not. I can only follow sources, and when these differ, I can't say for sure what is now correct.

Would be open for any changes regarding that!

File: UI/data/locale.ini [hunk 3]
```diff
@@ -65,10 +68,10 @@ Name=Svenska
 Name=Türkçe
 
 [zh-CN]
-Name=简体中文
+Name=中文简体
```
Comment (RytoEX) [inline UI/data/locale.ini:71]: I had this reviewed by native speakers, and the ones I asked agreed that the original text is better than the proposed change.

File: UI/data/locale.ini [hunk 4]
```diff
@@ -65,10 +68,10 @@ Name=Svenska
...
 
 [zh-TW]
-Name=繁體中文
```
Comment (RytoEX) [inline UI/data/locale.ini:71]: Based on feedback for the above proposed change for zh-CN, I'd guess this is also not an ideal change.

**Review by RytoEX — CHANGES_REQUESTED** [review]
I asked native Chinese speakers about the changes proposed for zh-CN and zh-TW.  See review comments.

All in all, in the absence of a publicly available standard, my recommendation is to let native/near-native speakers propose changes like this.  I don't think we need to proactively chase these down.

**Pull request closed without merging**